src/Collection/CollectionInterface.php

```diff
@@ -1038,7 +1038,7 @@ public function zip(iterable $items): CollectionInterface;
      * @param callable $callback The function to use for zipping the elements together.
      * @return self
      */
-    public function zipWith(iterable $items, callable $callback): CollectionInterface;
+    public function zipWith(iterable $items, iterable|callable $callback): CollectionInterface;
```

Comment:
The types for `$callaback` get weird since the method allows variable number of items with a callback as the last argument. I think it's better to just not add any types for the `$callback` argument.

Reply:
Another option is to move the `$callback` to the 1st position so that we can use proper variadic argument for items.

Reply:
> Another option is to move the `$callback` to the 1st position so that we can use proper variadic argument for items.

That's the PHP approach, but we can probably just omit the type. I recall struggling with this when adding types. Probably have a comment in the original PR.

Reply:
I got rid to the type for `$callback` for now.